### PR TITLE
Print policy violations only if any rule was violated (AST-66024)

### PR DESCRIPTION
**Pull request by sarahCx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Check if there was no role that was violated in the policy, the policy will not be displayed in the PR decoration and in console summary.

### References

https://checkmarx.atlassian.net/browse/AST-66024

### Testing

To console summery: the test verifies that when there are no policy violations, the message "policy management violation" will not be printed.
To PR Decoration: the test verifies that when there are no policy violations, the policy will not be send to github.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/result.go [hunk 1]
```diff
@@ -791,13 +791,22 @@ func writeConsoleSummary(summary *wrappers.ResultSummary, featureFlagsWrapper wr
...
 
 func printPoliciesSummary(summary *wrappers.ResultSummary) {
-	fmt.Printf(tableLine + "\n")
-	if summary.Policies.BreakBuild {
-		fmt.Printf("            Policy Management Violation - Break Build Enabled:                     \n")
-	} else {
-		fmt.Printf("            Policy Management Violation:                     \n")
-	}
+	hasViolations := false
 	if len(summary.Policies.Policies) > 0 {
```
Comment (AlvoBen) [inline internal/commands/result.go:795]: Why is this if necessary?
if len(summary.Policies.Policies) == 0, we will not iterate it in the for loop either way

File: internal/commands/result.go [hunk 2]
```diff
@@ -791,13 +791,22 @@ func writeConsoleSummary(summary *wrappers.ResultSummary, featureFlagsWrapper wr
...
+		for _, policy := range summary.Policies.Policies {
```
Comment (AlvoBen) [inline internal/commands/result.go:796]: please extract it to new methos - checkIfPolicyViolated(.....)

File: internal/commands/result.go [hunk 3]
```diff
@@ -791,13 +791,22 @@ func writeConsoleSummary(summary *wrappers.ResultSummary, featureFlagsWrapper wr
...
+			if len(policy.RulesViolated) > 0 {
+				hasViolations = true
+				break
+			}
+		}
+	}
+	if hasViolations {
```
Comment (AlvoBen) [inline internal/commands/result.go:803]: checkIfPolicyViolated(.....)

File: internal/commands/result.go [hunk 4]
```diff
@@ -807,8 +816,8 @@ func printPoliciesSummary(summary *wrappers.ResultSummary) {
 			}
 			fmt.Printf("\n")
 		}
+		fmt.Printf("\n")
```
Comment (AlvoBen) [inline internal/commands/result.go:819]: if there are no violation, this "\n" will not be printed, is this correct?
Before your changed, we always printed it

File: internal/commands/result.go (hunk 1 above)
Comment (sarahCx) [reply internal/commands/result.go:795]: This is the bug, the function get a policy without a roll violation.

File: internal/commands/result.go (hunk 4 above)
Comment (sarahCx) [reply internal/commands/result.go:819]: Yes, because before my changes always we printed the header, and now we didn't print anything.

**Pull request merged**